src/components/ProductTours/AdminOnboardingTours/useLearnerProgressTour.tsx

```diff
@@ -1,40 +1,99 @@
-import { ReactNode } from 'react';
+import { ReactNode, useCallback, useState } from 'react';
 import { sendEnterpriseTrackEvent } from '@edx/frontend-enterprise-utils';
 import { useIntl } from '@edx/frontend-platform/i18n';
+import { logError } from '@edx/frontend-platform/logging';
 
 import {
   ADMIN_TOUR_EVENT_NAMES,
-  ADMIN_TOUR_TARGETS,
+  TRACK_LEARNER_PROGRESS_TARGETS,
 } from './constants';
 
 import messages from './messages';
+import LmsApiService from '../../../data/services/LmsApiService';
+import { flowUuids } from '../../../config';
 
 interface TourStep {
   target: string;
   placement: 'right' | 'left' | 'top' | 'bottom';
-  title: ReactNode;
+  title?: ReactNode;
   body: ReactNode;
   onAdvance: () => void;
 }
 
 interface UseLearnerProgressTourProps {
   enterpriseSlug: string;
+  adminUuid: string;
+  aiButtonVisible: boolean;
 }
 
-const useLearnerProgressTour = ({ enterpriseSlug }: UseLearnerProgressTourProps): TourStep => {
-  // TO-DO: Handle in next ticket
-  const handleAdvanceTour = () => {
-    sendEnterpriseTrackEvent(enterpriseSlug, ADMIN_TOUR_EVENT_NAMES.LEARNER_PROGRESS_ADVANCE_EVENT_NAME);
-  };
+const useLearnerProgressTour = (
+  { enterpriseSlug, adminUuid, aiButtonVisible }: UseLearnerProgressTourProps,
+): Array<TourStep> => {
   const intl = useIntl();
+  const [stepIndex, setStepIndex] = useState(0);
+
+  const handleAdvanceTour = useCallback(() => {
+    const newIndex = stepIndex + 1;
+    sendEnterpriseTrackEvent(enterpriseSlug, ADMIN_TOUR_EVENT_NAMES.LEARNER_PROGRESS_ADVANCE_EVENT_NAME, { 'completed-step': newIndex });
+    setStepIndex(newIndex);
+  }, [enterpriseSlug, stepIndex]);
+
+  const handleEndTour = async () => {
+    try {
+      sendEnterpriseTrackEvent(enterpriseSlug, ADMIN_TOUR_EVENT_NAMES.LEARNER_PROGRESS_ADVANCE_EVENT_NAME);
+      await LmsApiService.updateCompletedTourFlows(adminUuid, flowUuids.TRACK_LEARNER_PROGRESS_UUID);
```

Comment:
Can we establish a pattern in this PR for storing flow-specific values like these, instead of hardcoding the LPR values?  Otherwise, it will fall to the person implementing the second flow to establish the pattern, and given we may have multiple people working in parallel on the next flows, establishing the pattern now will help avoid coordinating problems later.